pipeline-steps/deploy.groovy

```diff
@@ -35,4 +35,62 @@ def deploy_sh(Map args) {
   ) // conditionalStage
 }
 
+def deploy_maas() {
+  common.conditionalStage(
+    stage_name: 'Prepare MaaS Configs',
+    stage: {
+      withCredentials([
+        usernamePassword(
+          credentialsId: "dev_pubcloud_user_key",
+          usernameVariable: "PUBCLOUD_USERNAME",
+          passwordVariable: "PUBCLOUD_APIKEY"
+        ),
+        string(
+          credentialsId: "dev_pubcloud_tennant",
+          variable: "PUBCLOUD_TENANT_ID"
+        )
+      ]){
+        sh """
+          export ADMIN_PWD=\$(sudo grep keystone_auth_admin_password /etc/openstack_deploy/user_osa_secrets.yml  | awk '{print \$2}')
+          export HOSTNAME=\$(hostname)
+
+          sudo -E sh -c 'echo "
+rackspace_cloud_username: ${env.PUBCLOUD_USERNAME}
+rackspace_cloud_api_key: ${env.PUBCLOUD_APIKEY}
+rackspace_cloud_tenant_id: ${env.PUBCLOUD_TENANT_ID}
+maas_notification_plan: npTechnicalContactsEmail
+maas_fqdn_extension: .${env.NODE_NAME}
+memory_used_percentage_warning_threshold: 99.0
+memory_used_percentage_critical_threshold: 99.5
+net_max_speed: 1000
+lb_name: \${HOSTNAME}
+
+keystone_service_adminurl: http://172.29.236.200:5000/v3
+keystone_admin_user_name: admin
+keystone_admin_tenant_name: admin
+keystone_auth_admin_password: \${ADMIN_PWD}" >> /etc/openstack_deploy/user_rpco_variables_overrides.yml'
```

Comment:
I had a lot of problems before where these variables weren't getting picked up for some reason, and this was the only way I could get it to work. I'm not sure if it would still be an issue if we don't override them since that I made that job a long time ago, though, so maybe it's worth trying again without them.